### PR TITLE
QA issue - email validations misfiring when revisting page - RIP-119

### PR DESCRIPTION
**Pull request by autotelik**
_If you go back to the page and change the second email address and then continue, it doesn't validate the two fields match (although it does if you change the first email address)
Also if you change both emails and continue it shows a validation error that the emails don't match too._

Fixed by fixing method that checks if email fields have changed in the Form

### CODE REVIEW

**Comment by autotelik** [comment]
:eyes:

File: app/controllers/flood_risk_engine/enrollments/steps_controller.rb [hunk 1]
```diff
@@ -19,6 +19,7 @@ def show
         render :show, locals: locals
       end
 
+      # rubocop:disable Metrics/AbcSize
```
Comment (reggieb) [inline app/controllers/flood_risk_engine/enrollments/steps_controller.rb]: I think this needs to be re-enabled after the method. Alternatively, move the error handling out into a separate method.

Comment (autotelik) [inline app/controllers/flood_risk_engine/enrollments/steps_controller.rb]: agreed. disabled for single method

**Comment by reggieb** [comment]
A general comment about `app/forms/flood_risk_engine/steps/correspondence_contact_email_form.rb`.

The validations are called at the class level where the `locale_key` method is available directly. So lines like this:
```ruby
validates :email_address, presence: {
          message: I18n.t("#{CorrespondenceContactEmailForm.locale_key}.errors.email_address.blank")
        }
```
can be written as:
```ruby
validates :email_address, presence: {
          message: I18n.t("#{locale_key}.errors.email_address.blank")
        }
```

**Comment by reggieb** [comment]
:ship: it.

**Comment by autotelik** [comment]
r.e General comment .. this is not actually the case, I've tried that but will cause a no method error as something like the ReFormGroup class is self in this location

**Pull request merged**